linalg/src/frame/reduce/sum.rs

```diff
@@ -0,0 +1,60 @@
+#[cfg(test)]
+#[macro_use]
+pub mod test {
+    use crate::frame::reduce::ReduceKer;
+    use crate::LADatum;
+    use num_traits::{AsPrimitive, Float, Zero};
+    use proptest::test_runner::TestCaseResult;
+
+    #[macro_export]
+    macro_rules! sum_frame_tests {
+        ($cond:expr, $t: ty, $ker:ty) => {
+            proptest::proptest! {
+                #[test]
+                // Addition is not associative in f16 a + b + c != (a + b) + c because of precision
+                // issues. To avoid that, test vector should be aligned with kernel size (to avoid
+                // having to sum partial sums together).
```

Comment:
is this still true ? it is a bit misleading in that f32 is not associative either.